JX_DefineMainFile not functioning in embedded Windows app

Have embedded jxcore into a large, pre-existing Win32 application, by compiling with --shared_library and including the jx.dll binary in my solution. Though JX_Evaluate and JX_DefineExtension can correctly run js and hooks, the contents of JX_DefineMainFile are never executed.

Making a minimal command-line app that calls jxcore in the same way works perfectly fine, including the ...MainFile being run, so I'm not sure where the difference lies. Tracing the execution of JX_StartEngine in a debugger, I see a callstack like this:

engine->Start()
  node::Load
    JS_METHOD_CALL
      i::Execution::Call
        Invoke
          CALL_GENERATED_CODE
(the debugger doesn't correctly step into this, so it's as far as I got)

So it looks the engine correctly sees that main has a function object defined, but doesn't execute it properly.

Any pointers on how to dig into this more deeply, diagnose the problem, or otherwise figure out what I'm doing wrong?

No wait, nevermind, my environment is fine. The build was broken by a3fc681b2bb7691c491b82a13f87f983cb256c4f, for Visual C++ 2010 Express. I'll look into fixing that first. The error is:

```
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\Microsoft.Cpp.InvalidPlatform.Targets(23,7): error MSB8007: The Platform for project 'cares.vcxproj' is invalid.  Platform='x64'. You may be seeing this message because you are trying to build a project without a solution file, and have specified a non-default Platform that doesn't exist for this project. [D:\Env\cygwin64\home\dlong\dev\code\cpp\jxcore\deps\cares\cares.vcxproj]
...repeats 17 times...
```

--- a/src/node.cc
+++ b/src/node.cc
@@ -2258,7 +2258,7 @@ void SetupProcessObject(const int threadId, bool debug_worker) {
   // Check if the process has standard stream file descriptors, needed for
   // allocating a normal console object.
   JS_NAME_SET(process, JS_STRING_ID("hasStdFds"),
-              STD_TO_BOOLEAN(uv_guess_handle(1) == UV_UNKNOWN_HANDLE));
+              STD_TO_BOOLEAN(uv_guess_handle(1) != UV_UNKNOWN_HANDLE));
 
   JS_NAME_SET(process, JS_STRING_ID("isPackaged"),
               STD_TO_BOOLEAN(com->is_packaged_));

